src/app/models/vehicle-test-result-update.ts

```diff
@@ -0,0 +1,7 @@
+import { UserDetails } from '@app/models/user-details';
+import { TestResultModel } from '@app/models/test-result.model';
+
+export interface VehicleTestResultUpdate {
+  testResult: TestResultModel;
+  msUserDetails: UserDetails;
```

Comment:
Why have we got the msUserDetails in there? If i am not wrong, this is not part of the Techrecord or testResult object.
If we need it in the view it should probably be at the root of the store in a separate reducer so that it can be shared easily
```
{
  techRecord: [],
  loading: false,
  error: null,
  msUser: null,
  ...
}
```
it is worth having a look at the API spec with the swagger file.
https://github.com/dvsa/cvs-svc-technical-records/blob/develop/tests/resources/technical-records.json
https://github.com/dvsa/cvs-svc-test-results/blob/develop/tests/resources/test-results.json

Also why do we need a separate model to update the test result?
Can we just have a vehicle-test-result model?

Reply:
I think the problem here is back end expects a different object to allow savings. We have a singleton that will give us the user details as expected

Reply:
I don't remember seeing that as needed on the POST request and this won't be required I believe: https://github.com/dvsa/cvs-svc-test-results/blob/develop/tests/resources/test-results-post.json